main.go

```diff
@@ -75,11 +76,21 @@ func main() {
 	ctx := context.Background()
 	log := logger.NewLogger(ctx)
 	ctx = context.WithValue(ctx, logger.LoggerKey{}, log)
+	undoMaxprocs, maxprocsError := maxprocs.Set(maxprocs.Logger(func(msg string, keysAndValues ...interface{}) {
+		log.Klogr.WithValues("component", "maxprocs", "version", maxprocs.Version).V(2).Info(fmt.Sprintf(msg, keysAndValues...))
+	}))
+	defer undoMaxprocs()
+
+	if maxprocsError != nil {
+		log.Error(maxprocsError, "Failed to set GOMAXPROCS")
+		os.Exit(1)
```

Comment:
I don't think we should necessarily exit if we can't set `GOMAXPROCS` since that could result in a CrashLoopBackoff where there was none prior without this, but we should keep the error log.

Reply:
I agree. This would introduce a new problem. I'll fix that and push the changes.